### DISCUSSION TITLE
How do I change Keycloak 20.0.0 admin console URL in CentOS7?

### DISCUSSION THREAD

**Discussion by gitblanc** [keycloak/keycloak#22343] [Q&A] [answered]
I've been searching a lot about how do I change the default url of the admin console of Keycloak in CentOS7. My current URL is like this schema: https://my-domain/admin/master/console, so I want to change the admin part to something like https://my-domain/new-url/master/console Most sites say that you can modify the standalone.xml file, but I couldn't find that file in version 20.0.0. It seems that this file is not available anymore since version 14.

I hope that someone can help me to solve this big problem, because even I have read all the official documentation, I couldn't find anything.
You can check this same question on StackOverflow here -> https://stackoverflow.com/questions/76867645/how-do-i-change-keycloak-20-0-0-admin-console-url-in-centos7

**Comment by vmuzikar** [comment, marked as answer]
Unfortunately, selectively changing context root just for admin console is not supported OOTB. However, you should be able to configure a reverse proxy to do that for you. You can then configure the [hostname settings](https://www.keycloak.org/server/hostname#_administration_console) so Keycloak expects your new URL. Note that this settings doesn't actually affect the context root (or ports) that Keycloak uses, it just modifies the redirect URLs, links in the UI etc.

**Reply by gitblanc** [reply]
But in theory I have that edited on the keycloak.conf file. I have something like this:
````
#HTTP Host config
hostaname=mydomain
hostname-strict=false
hostname-strict-backchannel=true
http-relative-path=/
proxy=edge
health-enabled=true

#hostname-admin-url=https://mydomain/test/master/console
````
so shouldn't happen right?

**Reply by vmuzikar** [reply]
> so shouldn't happen right?

What do you mean? :)

**Reply by gitblanc** [reply]
I mean that if I have that previous config, when I enter the admin console, in the url shouldn't appear /admin/...

**Reply by vmuzikar** [reply]
No. As I stated, the hostname options only affects redirect URIs, links (to CSS, JS, ... files). Not the actual path Keycloak serves on the Admin Console. This is meant to work in combination with a reverse proxy.

**Reply by gitblanc** [reply]
Ahh, sorry for the confusion. I understand what you are saying now. Thanks a lot!
